src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -582,48 +584,52 @@ private void invoke(
             try {
                 TBase<TBase<?, ?>, TFieldIdEnum> wrappedResult = func.newResult();
                 func.setSuccess(wrappedResult, result);
-                respond(serializationFormat, seqId, func, wrappedResult, res);
+                respond(ctx, serializationFormat, seqId, func, wrappedResult, res);
             } catch (Throwable t) {
                 final TBase<TBase<?, ?>, TFieldIdEnum> exceptionResult = func.newResult();
                 if (func.setException(exceptionResult, t)) {
-                    respond(serializationFormat, seqId, func, exceptionResult, res);
+                    respond(ctx, serializationFormat, seqId, func, exceptionResult, res);
                 } else {
-                    respond(serializationFormat, seqId, func, t, res);
+                    respond(ctx, serializationFormat, seqId, func, t, res);
                 }
             }
         })).exceptionally(CompletionActions::log);
     }
 
     private static void handleException(
+            ServiceRequestContext ctx,
             SerializationFormat serializationFormat,
             int seqId, ThriftFunction func, Throwable cause, HttpResponseWriter res) {
 
         final TBase<TBase<?, ?>, TFieldIdEnum> result = func.newResult();
         if (func.setException(result, cause)) {
-            respond(serializationFormat, seqId, func, result, res);
+            respond(ctx, serializationFormat, seqId, func, result, res);
         } else {
-            respond(serializationFormat, seqId, func, cause, res);
+            respond(ctx, serializationFormat, seqId, func, cause, res);
         }
     }
 
-    private static void respond(SerializationFormat serializationFormat, int seqId,
+    private static void respond(ServiceRequestContext ctx,
+                                SerializationFormat serializationFormat, int seqId,
                                 ThriftFunction func, TBase<TBase<?, ?>, TFieldIdEnum> result,
                                 HttpResponseWriter res) {
         respond(serializationFormat,
-                encodeSuccess(serializationFormat, func, seqId, result),
+                encodeSuccess(ctx, serializationFormat, func, seqId, result),
                 res);
     }
 
-    private static void respond(SerializationFormat serializationFormat, int seqId,
+    private static void respond(ServiceRequestContext ctx,
```

Comment:
Good idea. Let me also replace the `ThriftFunction` parameters in `respond()` methods with `String methodName`.